app/app.go

```diff
@@ -101,8 +101,9 @@ func New(ctx context.Context, adminPort string) (*App, error) {
 		mux.HandleFunc("/debug/dump/memstats", dumpMemStats)
 
 		server := http.Server{
-			Addr:    ":" + adminPort,
-			Handler: mux,
+			Addr:              ":" + adminPort,
+			Handler:           mux,
+			ReadHeaderTimeout: 60 * time.Second,
```

Comment:
I'm not sure about two things:

 * Shouldn't we use "ReadTimeout", if safety is important here? I'm concerned that the same DOS attack would work in the body of the request as well
 * Ideally, this value should come from a config.

This solution is good enough to fix the security issue for now, but maybe the code owners will have an opinion about this

Reply:
I agree with both points.

"ReadTimeout" is interesting, but it might be more sensitive to the system being implemented. It's unusual for a header take too long to be received; a higher read timeout, however might affect synchronous API's or systems which are expected to receive very large input. Using "ReadHeaderTimeout" has more limited impact, so we can ship it quickly with lower risk all the same without disabling this linter check for other changesets.

Configuration is larger changeset with some room for discussion. I'll open another pull request which adds "ReadTimeout" and make both "ReadTimeout" and "ReadHeaderTimeout" configurable.